apps/etl-func/src/domain/interfaces/tokenizer.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from "zod";
+
+export interface TokenizerClient {
+  tokenize(pii: string): Promise<string>;
+}
+
+const messageTokenizedSchema = z.object({
+  fiscalCode: z.string().uuid().min(1),
+});
+
+export type messageTokenized = z.infer<typeof messageTokenizedSchema>;
+
+export const tokenize =
+  (message: { fiscalCode: string }) =>
+  async (client: TokenizerClient): Promise<messageTokenized> => {
+    const tokenizedFiscalCode = await client.tokenize(message.fiscalCode);
+    return { ...message, fiscalCode: tokenizedFiscalCode };
```

Comment:
Is it correct that we have logic inside `interfaces/` ?

Reply:
this is a domain logic specific for the tokenization of sensible information and I think it will be superfluous to create another file just for that. I think of this method as if it was a default method of the interface